source/applications/advanced/color_balance.py

```diff
@@ -0,0 +1,239 @@
+            f"The dimension value should be '3d' or '2d', got: '{dimension}'"
+        )
+
+    return settings
+
+
+def _capture_rgb(camera, settings_2d):
+    """Capture 2D RGB image.
+
+    Args:
+        camera: Zivid camera
+        settings_2d: 2D capture settings
+
+    Returns:
+        rgb: RGB image
+
+    """
+    frame_2d = camera.capture_2d(settings_2d)
+    image = frame_2d.image().to_array()
+    rgb = np.dstack([image["r"], image["g"], image["b"], image["a"]])
+
+    return rgb
+
+
+def _compute_mean_rgb(rgb, pixels):
+    """Compute mean RGB values.
+
+    Args:
+        rgb: RGB image
+        pixels: Number of central pixels for computation
+
+    Returns:
+        mean_color: RGB channel mean values
+
+    """
+    height = np.shape(rgb)[0]
+    width = np.shape(rgb)[1]
+    pixel_hight_start_index = int((height - pixels) / 2)
+    pixel_hight_end_index = int((height + pixels) / 2)
+    pixel_width_start_index = int((width - pixels) / 2)
+    pixel_width_end_index = int((width + pixels) / 2)
+    red = rgb[
+        pixel_hight_start_index:pixel_hight_end_index,
+        pixel_width_start_index:pixel_width_end_index,
+        0,
+    ]
+    green = rgb[
+        pixel_hight_start_index:pixel_hight_end_index,
+        pixel_width_start_index:pixel_width_end_index,
+        1,
+    ]
+    blue = rgb[
+        pixel_hight_start_index:pixel_hight_end_index,
+        pixel_width_start_index:pixel_width_end_index,
+        2,
+    ]
+    mean_red = np.mean(np.reshape(red, -1), dtype=np.float64)
+    mean_green = np.mean(np.reshape(green, -1), dtype=np.float64)
+    mean_blue = np.mean(np.reshape(blue, -1), dtype=np.float64)
+    mean_color = MeanColor(red=mean_red, green=mean_green, blue=mean_blue)
+
+    return mean_color
+
+
+def _apply_color_balance(rgb, red_balance, blue_balance):
+    """Apply color balance to RGB image.
+
+    Args:
+        rgb: Input RGB image
+        red_balance: Red balance
+        blue_balance: Blue balance
+
+    Returns:
+        corrected_image: RGB image after color balance
+
+    """
+    default_red_balance = zivid.Settings().red_balance
+    default_blue_balance = zivid.Settings().blue_balance
+    corrected_rgb = np.copy(rgb)
+    corrected_rgb[:, :, 0] = rgb[:, :, 0] * red_balance / default_red_balance
+    corrected_rgb[:, :, 2] = rgb[:, :, 2] * blue_balance / default_blue_balance
+
+    return corrected_rgb
+
+
+def _color_balance_calibration(camera, settings_3d):
+    """Balance color for RGB image by taking images of white surface (piece of paper, wall, etc.) in a loop.
+
+    Args:
+        camera: Zivid camera
+        settings_3d: 3D capture settings
+
+    Returns:
+        corrected_red_balance: Corrected red balance
+        corrected_blue_balance: Corrected blue balance
+
+    """
+    print(f"Starting color balance calibration")
+    corrected_red_balance = 1.0
+    corrected_blue_balance = 1.0
+    settings_list = []
+    settings_list.append(settings_3d)
```

Comment:
This can be simplified to: `settings_list = [settings_3d]`

Reply:
Done